lib/persist.js

```diff
@@ -1,50 +1,66 @@
 require("dotenv").config();
-const redis = require("redis").createClient(process.env.REDIS_URL);
-const { promisify } = require("util");
-const get = promisify(redis.get).bind(redis); // this maps redis.get(key, callback(results)) to a promise-compatible function
+
 const moment = require("moment");
 const { getCurrentEmojis } = require("./slack");
 
+const makeRedisClient = async () => {
+  const redis = require("redis").createClient({
+    url: process.env.REDIS_URL,
+  });
+  redis.on("error", (err) => console.log("Redis Client Error", err));
+  await redis.connect();
+
+  return redis;
+};
+
 const FORBIDDEN_DAYS_OF_WEEK = ["Saturday", "Sunday"];
 
-const getLastKnownEmoji = async (todayIso) => {
```

Comment:
this was cursed code. previously i was looking up 7 days of history to try to find the last emoji list. now we always store the list in `last_known` so i can just directly look for it.